File: common/src/main/java/dev/number6/db/model/converter/LocalDateConverter.java
package dev.number6.db.model.converter;

import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConverter;

import java.time.LocalDate;

public class LocalDateConverter implements DynamoDBTypeConverter<String, LocalDate> {

    @Override
    public String convert(LocalDate object) {
        return object.toString();
    }

    @Override
    public LocalDate unconvert(String object) {
        return LocalDate.parse(object);
    }
}

File: common/src/main/kotlin/dev/number6/db/model/converter/LocalDateConverter.kt
package dev.number6.db.model.converter

import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConverter
import java.time.LocalDate

class LocalDateConverter : DynamoDBTypeConverter<String, LocalDate> {
    override fun convert(date: LocalDate): String {
        return date.toString()
    }

    override fun unconvert(date: String): LocalDate {
        return LocalDate.parse(date)
    }
}